### DISCUSSION TITLE
Minesweeper Nillion

### DISCUSSION THREAD

**Discussion by envoy1084** [NillionNetwork/discussions#42] [Show and tell]
### Show and tell project type

Builder Bounty Submission

### Github Repo Link

https://github.com/Envoy-VC/minesweeper-nillion

### Video Walkthough Link

https://www.youtube.com/watch?v=sRLVX6E1iUQ

### Project Description

```bash
This is a simple implementation of the classic game Minesweeper using Nillion Blind Computation.

The Game is inspired from the OG Windows XP Minesweeper game and is played in a 2D grid where the player must reveal all the cells that do not contain a mine.

The player can reveal a cell by clicking on it. If the cell contains a mine, the game is over. If the cell does not contain a mine, the cell will reveal a number that indicates the number of mines in the neighboring cells. 

The player can also flag a cell by ⌥ + CLICK on it. The player wins the game when all the cells that do not contain a mine are revealed.
```

### What problems does your project solve? How does it preserve privacy for users?

```bash
The project preserves privacy for users by hiding the mine locations from the other party and comping on finding how many adjacent mines are on that tile.
```

### How does the project use Nillion? Describe and link to any Nada programs

```bash
The nada programs has two parties:

1. Party1: The party who places mines.
2. Party2: The party who minesweeps.

The program takes total of 152 inputs, divided into two parts:

1. Mine Locations: Total of 75 mines divided as x , y values so total of 150 values.
2. User Move: The user move location as location-0 and location-1 as x , y value.

The main logic of Minesweeper game is to check the number adjacent mines and do the following:

1. If the cell has a mine, the game is over.
2. If the cell has no mine, reveal the cell and check the number of mines in the neighboring cells.
3. If the cell has no mine and no neighboring mines, reveal all the neighboring cells.
4. If the cell has no mine and has neighboring mines, reveal the cell and show the number of neighboring mines.

The nada program calculates the number of adjacent mines on the grid to that location and also checks if the location has a mine.

The program returns two outputs:

1. adjacent_mines: The number of mines in the neighboring cells.
2. game_over: If the cell has a mine, the game is over.

Entire Program can be found at: https://github.com/Envoy-VC/minesweeper-nillion/blob/main/packages/programs/src/main.py
```

### Is there anything else you want to share?

Private equality is throwing some errors during running the programs. The program is built successfully but during running the program throws `not enough particles error`

### Optional - Link your project and team members' social handles

https://twitter.com/Envoy_1084

### Optional - Team ETH Address(es)

- 0xBF4979305B43B0eB5Bb6a5C67ffB89408803d3e1

**Comment by oceans404** [comment]
Congrats again @Envoy-VC on winning the Minesweeper bounty! 

https://github.com/orgs/NillionNetwork/projects/3/views/1?pane=issue&itemId=63608769

**Reply by Sahnis89** [reply]
Minesweeper Nillion

**Comment by SadOps** [comment]
does nillion expose users data ?

**Comment by Sahnis89** [comment]
No not at all

User safety first



Thanks & Regards

---------- Forwarded message ---------
From: SadOps ***@***.***>
Date: Sun, 16 Jun 2024, 9:37 am
Subject: Re: [NillionNetwork/discussions] Minesweeper Nillion (Discussion
#42)
To: NillionNetwork/discussions ***@***.***>
Cc: Sahnis89 ***@***.***>, Comment ***@***.***>


does nillion expose users data ?

—
Reply to this email directly, view it on GitHub
<https://github.com/orgs/NillionNetwork/discussions/42#discussioncomment-9784750>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/BBN34PF2OES2XS4XSHGPKXDZHUFRFAVCNFSM6AAAAABH42BQEGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TOOBUG42TA>
.
You are receiving this because you commented.Message ID:
***@***.***>
